WorkspaceServer/Kernel/CodeSubmissionProcessors.cs

```diff
@@ -32,25 +33,37 @@ public void Register(ICodeSubmissionProcessor processor)
 
         public async Task<SubmitCode> ProcessAsync(SubmitCode codeSubmission)
         {
-            var lines = new Queue<string>( codeSubmission.Value.Split(new[] {"\r\n", "\n"}, StringSplitOptions.None));
-            var unhandledLines = new Queue<string>();
-            while (lines.Count > 0)
+            try
             {
-                var currentLine = lines.Dequeue();
-                var result = _parser.Parse(currentLine);
-
-                if (result.CommandResult != null && _processors.TryGetValue(result.CommandResult.Command, out var processor))
-                {
-                    await _parser.InvokeAsync(result);
-                    var newSubmission =  await processor.ProcessAsync(new SubmitCode(string.Join("\n", lines), codeSubmission.Id, codeSubmission.ParentId));
-                    lines = new Queue<string>(newSubmission.Value.Split(new[] { "\r\n", "\n" }, StringSplitOptions.None));
-                }
-                else
+                var lines = new Queue<string>(codeSubmission.Value.Split(new[] {"\r\n", "\n"},
+                    StringSplitOptions.None));
+                var unhandledLines = new Queue<string>();
+                while (lines.Count > 0)
                 {
-                    unhandledLines.Enqueue(currentLine);
+                    var currentLine = lines.Dequeue();
+                    var result = _parser.Parse(currentLine);
+
+                    if (result.CommandResult != null &&
+                        _processors.TryGetValue(result.CommandResult.Command, out var processor))
+                    {
+                        await _parser.InvokeAsync(result);
+                        var newSubmission = await processor.ProcessAsync(new SubmitCode(string.Join("\n", lines),
+                            codeSubmission.Id, codeSubmission.ParentId));
+                        lines = new Queue<string>(newSubmission.Value.Split(new[] {"\r\n", "\n"},
+                            StringSplitOptions.None));
+                    }
+                    else
+                    {
+                        unhandledLines.Enqueue(currentLine);
+                    }
                 }
+
+                return new SubmitCode(string.Join("\n", unhandledLines), codeSubmission.Id, codeSubmission.ParentId);
+            }
+            catch (Exception e)
```

Comment:
This seems overly broad. We should ideally only catch exceptions that we understand, as this might otherwise mask some of our own bugs.

Reply:
that is wrapping all exceptions so we capture the exception and the code submission